universal/src/crypto/certificate.cpp

```diff
@@ -86,6 +86,38 @@ CertificatesChain LoadCertificatesChainFromString(std::string_view chain) {
     return certificates;
 }
 
+std::string Certificate::GetSubject() const {
+    X509* x509 = GetNative();
+    if (!x509) {
+        throw KeyParseError(FormatSslError("Invalid certificate"));
+    }
+
+    X509_NAME* subject_name = X509_get_subject_name(x509);
+    if (!subject_name) {
+        throw KeyParseError(FormatSslError("Failed to get subject name from certificate"));
+    }
+
+    // Используем BIO для более гибкого форматирования
+    BIO* bio = BIO_new(BIO_s_mem());
+    if (!bio) {
+        throw KeyParseError(FormatSslError("Failed to create BIO"));
+    }
+
+    // Флаги форматирования можно настроить по вкусу
+    const int flags = XN_FLAG_RFC2253;  // компактная строка, как в DN
+    if (X509_NAME_print_ex(bio, subject_name, 0, flags) < 0) {
+        BIO_free(bio);
+        throw KeyParseError(FormatSslError("Failed to print subject name"));
+    }
+
+    char* data = nullptr;
+    long len = BIO_get_mem_data(bio, &data);
+    std::string result(data, len);
```

Comment:
The constructor may throw (although it's rare - it's still a possibility) which will leak bio

Reply:
It's better to use RAII as the reasoning about the exception safety is quite complex